File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length, sum = 1;
        int[] ans = new int[n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                if(i == j) continue;
                sum *= nums[j];
            }
            ans[i] = sum;
            sum = 1;
        }
        return ans;
    }
}  

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> ans;
        unordered_map<int, int> hm;
        
        int n = nums.size();
        for(int i = 0; i < n; i++) {
            auto it = hm.find(nums[i]); 
            if(it != hm.end()) {
                it->second += 1; 
            } else {
                hm.insert({nums[i], 1});
            }
        }

//Điều kiện là chỉ lấy đủ size của map dựa trên k và 
        for(auto &pair : hm){
            auto key = pair.first, value = pair.second;
            int size = hm.size();
            if(size - k == 0){
                ans.push_back(key);
            } else if(size - k > 0){
                if(value >= k){
                    ans.push_back(key);
                }
            }
        }
        return ans;
    }
};
